spring-cloud-dataflow-platform-cloudfoundry/src/main/java/org/springframework/cloud/dataflow/server/config/cloudfoundry/CloudFoundryDataFlowServerConfiguration.java

```diff
@@ -45,14 +46,17 @@ public CloudFoundryDeploymentProperties taskDeploymentProperties() {
 
 	@Bean
 	public CloudFoundryServerConfigurationProperties cloudFoundryServerConfigurationProperties() {
-		return new CloudFoundryServerConfigurationProperties();
+		CloudFoundryServerConfigurationProperties cloudFoundryServerConfigurationProperties = new CloudFoundryServerConfigurationProperties();
+		return cloudFoundryServerConfigurationProperties;
 	}
 
-	@PostConstruct
-	public void afterPropertiesSet() {
-		if (cloudFoundryServerConfigurationProperties().isDebugReactor()) {
+	// Replaced PostConstruct with handling ContextRefreshedEvent allows for the
```

Comment:
1. I think this comment would be better served as a PR comment but I think its straight forward enough that we do not need it in the code. 

2. So this approach works. I wanted to show what it would look like using @ConditionOnProperty and nested config to do the same. 

```java
@ConditionalOnCloudPlatform(CloudPlatform.CLOUD_FOUNDRY)
@Configuration(proxyBeanMethods = false)
@EnableConfigurationProperties(CloudFoundryServerConfigurationProperties.class)
public class CloudFoundryDataFlowServerConfiguration {
	
	@Bean
	@ConfigurationProperties(prefix = CloudFoundryConnectionProperties.CLOUDFOUNDRY_PROPERTIES + ".task")
	public CloudFoundryDeploymentProperties taskDeploymentProperties() {
		return new CloudFoundryDeploymentProperties();
	}

	@Configuration(proxyBeanMethods = false)
	@ConditionalOnProperty(prefix = "spring.cloud.dataflow.server.cloudfoundry", name="is-debug-reactor", havingValue = "true")
	static class HooksDebugConfig {
		@PostConstruct
		void enableHooksDebug() {
			Hooks.onOperatorDebug();
		}
	}
}
```

I am not in ❤️ w/ the need to have the PostConstruct in the nested config. The listener approach may be cleaner. 

3. Either way, we can move the creation of the config props to `@EnableConfigurationProperties`.

Reply:
I decided on the listener since the `enableHooksDebug()` call isn't required during the bean init process.